internal/storage/image.go

```diff
@@ -638,7 +638,12 @@ func pullImageChild() {
 		os.Exit(1)
 	}
 
-	output <- pullImageOutputItem{Result: transports.ImageName(destRef), Name: canonicalRef.Name(), Digest: canonicalRef.Digest()}
+	rawCanonical, ok := canonicalRef.Raw().(reference.Canonical)
+	if !ok {
+		fmt.Fprintf(os.Stderr, "Returned reference %v is not canonical", canonicalRef.Raw().String())
+		os.Exit(1)
```

Comment:
is this a fatal programming error? like can cri-o not recover at all? if so I think we should `panic()` here. If not, we should find a way to error the image pull

Reply:
See above, this is in a subprocess and this is the currently-designed error reporting mechanism.